### PR TITLE
Add documentation about GPU features

### PR DESCRIPTION
**Pull request by amontoison**
I updated some functions in CUDA.jl to make it working with Krylov methods (https://github.com/JuliaGPU/CUDA.jl/pull/540).

### CODE REVIEW

File: docs/src/gpu.md [hunk 1]
```diff
@@ -36,3 +36,52 @@ x, stats = lsmr(A_gpu, b_gpu)
 ```
 
 Optimized operator-vector products that exploit GPU features can be also used by means of linear operators.
+
+Preconditioners, especially incomplete Cholesky or Incomplete LU factorizations with triangular solves,
```
Comment (dpo) [inline docs/src/gpu.md:40]:
```suggestion
Preconditioners, especially incomplete Cholesky or Incomplete LU factorizations that involve triangular solves,
```

File: docs/src/gpu.md [hunk 2]
```diff
@@ -36,3 +36,52 @@ x, stats = lsmr(A_gpu, b_gpu)
...
+can be performed directly on GPU with efficient operators that take advantage of CUSPARSE routines.
```
Comment (dpo) [inline docs/src/gpu.md:41]:
```suggestion
can be applied directly on GPU thanks to efficient operators that take advantage of CUSPARSE routines.
```

File: docs/src/gpu.md [hunk 3]
```diff
@@ -36,3 +36,52 @@ x, stats = lsmr(A_gpu, b_gpu)
...
+
+```julia
```
Comment (dpo) [inline docs/src/gpu.md:43]:
```suggestion
### Example with a symmetric positive-definite system

```julia
```

File: docs/src/gpu.md [hunk 4]
```diff
@@ -36,3 +36,52 @@ x, stats = lsmr(A_gpu, b_gpu)
...
+
+# Operator that model P⁻¹
+y = similar(b_gpu); n = length(b_gpu); T = eltype(b_gpu)
+opM = LinearOperator(T, n, n, true, true, x -> ldiv!(y, P, x))
+
+# Solve a symmetric positive definite system with an incomplete Cholesky preconditioner on GPU
+(x, stats) = cg(A_gpu, b_gpu, M=opM)
+```
+
+```julia
```
Comment (dpo) [inline docs/src/gpu.md:66]:
```suggestion
### Example with a general square system

```julia
```

**Review by dpo — CHANGES_REQUESTED** [review]
Looks great. Thanks.

**Comment by amontoison** [comment]
Thanks Dominique ! Can we build the documentation without Travis (https://travis-ci.com/github/JuliaSmoothOptimizers/Krylov.jl) ?

**Comment by dpo** [comment]
I think we need to move to GitHub Actions. Have a look at the RipQP.jl repo, where Abel and Geoffroy are currently working on it.

File: docs/src/gpu.md [hunk 5]
```diff
@@ -36,3 +36,56 @@ x, stats = lsmr(A_gpu, b_gpu)
...
+using CUDA, Krylov
+using CUDA.CUSPARSE, SparseArrays
+
+# LLᵀ ≈ A
+P = ic02(A_gpu, 'O')
+
+# Solve Py = x
+function ldiv!(y, P, x)
+  y .= x
+  sv2!('T', 'L', 1.0, P, y, 'O')
```
Comment (amontoison) [inline docs/src/gpu.md:55]: sv2!('T', 'U', 1.0, P, y, 'O')

**Comment by amontoison** [comment]
Tim merged my PR for CUDA.jl this morning for information.

**Comment by dpo** [comment]
I'm not entirely sure but perhaps this error is blocking doc deployment: https://github.com/JuliaSmoothOptimizers/Krylov.jl/runs/1390984166?check_suite_focus=true#step:5:18

**Comment by amontoison** [comment]
The documentation should be deployed when the PR is merged. I think it's the correct behaviour. :crossed_fingers:

**Comment by amontoison** [comment]
I checked with older pull requests and we also have this error with Travis (https://travis-ci.com/github/JuliaSmoothOptimizers/Krylov.jl/jobs/405315872à

**Comment by dpo** [comment]
Thank you!

**Pull request merged**